parlai/core/metrics.py

```diff
@@ -34,6 +34,100 @@
+    "exs": MetricDisplayData(
+        "Examples", "Number of examples processed since last print"
+    ),
+    "f1": MetricDisplayData(
+        "F1", "Unigram F1 overlap, under a standardized (model-independent) tokenizer"
+    ),
+    "gnorm": MetricDisplayData("Gradient Norm", "Gradient norm"),
+    "gpu_mem": MetricDisplayData(
+        "GPU Memory",
+        "Fraction of GPU memory used. May slightly underestimate true value.",
+    ),
+    "hits@1": MetricDisplayData(
+        "Hits@1", "Fraction of correct choices in 1 guess. (Similar to recall@K)"
+    ),
+    "hits@5": MetricDisplayData(
+        "Hits@5", "Fraction of correct choices in 5 guesses. (Similar to recall@K)"
+    ),
+    "interdistinct-1": MetricDisplayData(
+        "Interdistinct-1", "Fraction of n-grams unique across _all_ generations"
+    ),
+    "interdistinct-2": MetricDisplayData(
+        "Interdistinct-1", "Fraction of n-grams unique across _all_ generations"
+    ),
+    "intradistinct-1": MetricDisplayData(
+        "Intradictinct-1", "Fraction of n-grams unique _within_ each utterance"
+    ),
+    "intradictinct-2": MetricDisplayData(
+        "Intradictinct-2", "Fraction of n-grams unique _within_ each utterance"
+    ),
+    "jga": MetricDisplayData("Joint Goal Accuracy", "Joint Goal Accuracy"),
+    "llen": MetricDisplayData(
+        "Label Length", "Average length of label in number of tokens"
+    ),
+    "loss": MetricDisplayData("Loss", "Loss"),
+    "lr": MetricDisplayData("Learning Rate", "The most recent learning rate applied"),
+    "ltpb": MetricDisplayData("Label Tokens Per Batch", "Label tokens per batch"),
+    "ltps": MetricDisplayData("Label Tokens Per Second", "Label tokens per second"),
+    "ltrunc": MetricDisplayData(
+        "Label Truncation", "Fraction of samples with some label truncation"
+    ),
+    "ltrunclen": MetricDisplayData(
+        "Label Truncation Length", "Average length of label tokens truncated"
+    ),
+    "rouge-1": MetricDisplayData("ROUGE-1", "ROUGE metrics"),
+    "rouge-2": MetricDisplayData("ROUGE-2", "ROUGE metrics"),
+    "rouge-L": MetricDisplayData("ROUGE-L", "ROUGE metrics"),
+    "token_acc": MetricDisplayData(
+        "Token Accuracy", "Token-wise accuracy (generative only)"
+    ),
+    "token_em": MetricDisplayData(
+        "Token Exact Match",
+        "Utterance-level token accuracy. Roughly corresponds to perfection under greedy search (generative only)",
+    ),
+    "total_train_updates": MetricDisplayData(
+        "Total Train Updates", "Number of SGD steps taken across all batches"
+    ),
+    "tpb": MetricDisplayData(
+        "Tokens Per Batch", "Total tokens (context + label) per batch"
+    ),
+    "tps": MetricDisplayData(
+        "Tokens Per Second", "Total tokens (context + label) per second"
+    ),
+    "ups": MetricDisplayData("Updates Per Second", "Updates per second (approximate)"),
+}
+
+
+def get_metric_display_data(metric: str) -> MetricDisplayData:
```

Comment:
maybe as a utility of MetricsDisplayData

Reply:
It's kind of nice to keep this functional, though, since there isn't any state we should be keeping around. Also, it needs access to METRICS_DISPLAY_DATA which I think makes more sense scoped to the namespace than a class. What's the advantage you see from putting it in MetricDisplayData?

Reply:
I was thinking of a classmethod (and maybe the global too), just to keep everything in a tight namespace.

Reply:
lol prolly the global can't be in there so long as it's contained...

Reply:
Yeah I think if we went that path, the metrics/titles/descriptions would live in a separate json/yaml file. And we'd have a separate function that loads them up as `MetricDisplayData`s into a global. But then there'd be a disconnect between the source of truth and the global which is a little weird. I guess we could make `MetricDisplayData` a singleton and load them up on instantiation, but then we have to instantiate an object just to get this static list of strings which feels heavy.

Another way to keep them in a tight namespace would be to just create a `metrics_list.py` module.

Idk let me know if any of those options sound better, I see plenty of advantages and disadvantages to each so not super opinionated lol